src/server/actions/study-job.actions.ts

```diff
@@ -104,27 +110,83 @@ export const dataForJobAction = memberAction(async (studyJobIdentifier) => {
     return { jobInfo, manifest }
 }, z.string())
 
-export const latestJobForStudyAction = memberAction(async (studyId) => {
+export const dataForStudyDocumentsAction = async (studyId: string) => {
+    // Fetch study information
+    const studyInfo = await db
+        .selectFrom('study')
+        .innerJoin('member', 'study.memberId', 'member.id')
+        .select([
+            'study.id as studyId',
+            'study.title as studyTitle',
+            'study.descriptionDocPath',
+            'study.irbDocPath',
+            'study.agreementDocPath',
+            'member.identifier as memberIdentifier',
+        ])
+        .where('study.id', '=', studyId)
+        .executeTakeFirst()
+
+    if (!studyInfo) {
+        return null
+    }
+
+    // Prepare document list
+    const documents = []
+
+    if (studyInfo.descriptionDocPath) {
+        documents.push({
+            name: studyInfo.descriptionDocPath || 'Description Document',
+            path: studyInfo.descriptionDocPath,
+        })
+    }
+
+    if (studyInfo.irbDocPath) {
+        documents.push({
+            name: studyInfo.irbDocPath || 'IRB Document',
+            path: studyInfo.irbDocPath,
+        })
+    }
+
+    if (studyInfo.agreementDocPath) {
+        documents.push({
+            name: studyInfo.agreementDocPath || 'Agreement Document',
+            path: studyInfo.agreementDocPath,
+        })
+    }
```

Comment:
It doesn't seem like these if statements and object assignments are valid to me.  You're checking if the path is set, and then if it is you're pushing an object with the path for both name and path properties.  The name has a `||` but that will always be true because of the surrounding `if` statement.

Reply:
instead of all the `if` statements, I think you could use an array with filter/Boolean:
```
return {
  studyInfo,
  documents: [studyInfo.agreementDocPath, studyInfo.descriptionDocPath,studyInfo.irbDocPath].filter(Boolean)
} 
```
The Boolean will cast each to a bool and filter will remove any that are not true